modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java

```diff
@@ -120,15 +106,70 @@ public Set<Integer> getLivenessCheckPortNumbers() {
 
     @Override
     protected void configure() {
-        boolean emptyAdminPassword = this.adminPassword == null || this.adminPassword.isEmpty();
-        String neo4jAuth = emptyAdminPassword ? "none" : String.format(AUTH_FORMAT, this.adminPassword);
-        addEnv("NEO4J_AUTH", neo4jAuth);
+        configureAuth();
+        configureLabsPlugins();
+        configureExposedPorts();
+        configureWaitStrategy();
```

Comment:
Wait, I thought we wanted to move the defaults into the constructor (e.g., see `KafkaContainer`), to be explicitly friendly to user overrides as much as possible?

I know that we sometimes need to react dynamically to user configuration as much as possible, that's what `configure()` can then be used for.

Reply:
The problem that occurred to me was that e.g.
```
new Neo4jContainer<>(...)
  .withExposedPorts(7687);
```
would override things that we already set in the constructor.
So it has to get moved into the `configure()` method to allow me to respect `GenericContainer`-based user definitions. Otherwise we would expect e.g. that the two (three) ports are exposed by default.
The same applies to the `NEO4J_AUTH` environment variable. If we would set this fixed in the constructor, we would need to check again in the `configure()` method for overrides. This would put the logic into two places.
Since I had already moved the exposed ports and wait strategy, I wanted to keep it in the same spot. If you see any downside of this, I am happy to adjust it.

Reply:
I think your auto-configuration approach makes sense because you only apply the defaults if they have not been already set by the user. In this case, I was primarily concerned about pattern consistency across module implementations.

The only problem I see with the current approach is the disability of the user to set the startup timeout of the default wait strategy, a use case we have seen the need for in the past (sometimes depending on CI infrastructure even). So if we already set the `WaitStrategy` object in the constructor, this should help.

> If we would set this fixed in the constructor, we would need to check again in the configure() method for overrides. This would put the logic into two places.

Why would we need to check again? We would not touch those in the `configure()` method anymore. Just apply the default configs in the constructor.

/cc @eddumelendez, since this is a good opportunity to think about our general pattern for default configs.

Reply:
> since this is a good opportunity to think about our general pattern for default configs.

Yes

Regarding the startup timeout, we had a similar issue with PulsarContainer, see [this commit](https://github.com/testcontainers/testcontainers-java/commit/405ddb7e39443993f8988c4c5782727f0d0b5cc5). We should offer that flexibility.

Reply:
I completely agree on this. 
Wait strategy and default exposed ports are now back in the constructor while the `configure()` method will take care of user-made changes afterwards.
Also there are now some more comments in place to explain the "why".